src/lang/Class.h

```diff
@@ -60,20 +62,20 @@ class BLUEMEILIB_API Class
 	FieldMap m_fields;
 };
 
+
 //��ȡ������
-#define CLASS(className)\
-    Class(_T2STR(className),className::createObject)
-#define NEW_CLASS(className)\
-    new Class(_T2STR(className),className::createObject)
+#define CLASS(Cls) \
+    Class(_T2STR(Cls),Cls::createObject)
+#define NEW_CLASS(Cls) \
+    new Class(_T2STR(Cls),Cls::createObject)
 
 
 //�ɶ�̬�����������
-#define DECLARE_DCLASS(className) \
-    typedef className Self;                                                   \
+#define DECLARE_DCLASS(Cls)                                                   \
+    typedef Cls Self;                                                         \
     static Class* thisClass(){                                                \
-        static Class* cls = Class::registerClass(_T2STR(className),           \
-                                                 Self::createObject,          \
-                                                 __super::thisClass());       \
+        static Class* cls = Class::registerClass(_T2STR(Cls),                 \
+                Self::createObject, baseof(Cls)::thisClass());                \
```

Comment:
use baseof() instead of __super

about tr2::direct_bases please see: http://stackoverflow.com/questions/15158458/using-tr2direct-bases-get-nth-element-of-result